tests/utils.py

```diff
@@ -1,9 +1,89 @@
+import inspect
 from datetime import timedelta
 
+from django.apps import apps
+from factory.django import DjangoModelFactory
+
+from . import factories
+
 
 def minutes_ago(now, minutes):
     return now - timedelta(minutes=minutes)
 
 
 def seconds_ago(now, seconds):
     return now - timedelta(seconds=seconds)
+
+
+_SKIP_APP_LABELS = {"auth", "contenttypes", "sessions", "social_django"}
+
+_MODEL_TO_FACTORY = {
+    f._meta.model: f
+    for f in vars(factories).values()
+    if inspect.isclass(f) and issubclass(f, DjangoModelFactory)
+}
+
+
+def get_models():
+    """
+    Return a list of all installed, project-specific models.
+    """
+    return [
+        model
+        for model in apps.get_models()
+        if model._meta.app_label not in _SKIP_APP_LABELS
+    ]
+
+
+class MissingFactoryError(Exception):
+    pass
+
+
+def get_factory(model):
+    try:
+        return _MODEL_TO_FACTORY[model]
+    except KeyError:  # pragma: no cover
+        raise MissingFactoryError(f"{model._meta.label} is missing a factory")
+
+
+def get_paired_fields(model, lsuffix, rsuffix, exclude=None):
```

Comment:
I'm going to put these functions into `tests/paired_fields_utils.py` -- i.e. with the tests -- because `get_all_paired_fields` returns tuples that contain factories. If it returned tuples that contained models, then I'd put most of these functions in, say, `jobserver/models/paired_fields_utils.py`.

George mentioned that utils modules are untidy, so I've created #4108.